File: Problem-010/solution.js
function isPrime(num) {
    if (num < 2) return false;
    if (num === 2) return true;
    if (num % 2 === 0) return false;

    for (let i = 3; i * i <= num; i += 2) {
        if (num % i === 0) {
            return false;
        }
    }

    return true;
}

function primeSummation(limit) {
    let sum = 0;

    for (let i = 2; i < limit; i++) {
        if (isPrime(i)) {
            sum += i;
        }
    }

    return sum;
}

File: Problem-010/solution.py
def is_prime(num):
    if num < 2:
        return False
    if num == 2:
        return True
    if num % 2 == 0:
        return False

    i = 3
    while i * i <= num:
        if num % i == 0:
            return False
        i += 2

    return True


def summation_of_primes(limit):
    total = 0

    for i in range(2, limit):
        if is_prime(i):
            total += i

    return total
